### PR TITLE
feat(user): implement invitations api

### PR DESCRIPTION
**Pull request by racnan**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
New APIs to list of invitations for a user.

### Additional Changes

- [X] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
To show user list of invitations.

## How did you test it?
```sh
curl --location 'localhost:8080/user/list/invitation' \
--header 'Authorization: Bearer <JWT>' \
```

Response 
```json
[
    {
        "entity_id": "juspay2",
        "entity_type": "merchant",
        "entity_name": null,
        "role_id": "merchant_view_only"
    }
]
```

## Checklist

- [X] I formatted the code `cargo +nightly fmt --all`
- [X] I addressed lints thrown by `cargo clippy`
- [X] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/db/user_role.rs [hunk 1]
```diff
@@ -101,6 +105,8 @@ pub trait UserRoleInterface {
         profile_id: Option<&id_type::ProfileId>,
         entity_id: Option<&String>,
         version: Option<enums::UserRoleVersion>,
+        status: Option<UserStatus>,
+        limit: Option<u32>,
```
Comment (ThisIsMani) [inline crates/router/src/db/user_role.rs:109]: Create a type instead.

File: crates/router/src/core/user.rs [hunk 2]
```diff
@@ -2289,15 +2289,13 @@ pub async fn list_orgs_for_user(
...
         )
         .await
         .change_context(UserErrors::InternalServerError)?
         .into_iter()
-        .filter_map(|user_role| {
-            (user_role.status == UserStatus::Active)
-                .then_some(user_role.org_id)
-                .flatten()
-        })
+        .filter_map(|user_role| user_role.org_id)
```
Comment (ThisIsMani) [inline crates/router/src/core/user.rs:2298]: We should throw 500 if we don't find org_id in the user_role.

File: crates/router/src/core/user.rs [hunk 3]
```diff
@@ -2355,15 +2353,13 @@ pub async fn list_merchants_for_user_in_org(
...
             )
             .await
             .change_context(UserErrors::InternalServerError)?
             .into_iter()
-            .filter_map(|user_role| {
-                (user_role.status == UserStatus::Active)
-                    .then_some(user_role.merchant_id)
-                    .flatten()
-            })
+            .filter_map(|user_role| user_role.merchant_id)
```
Comment (ThisIsMani) [inline crates/router/src/core/user.rs:2362]: Here as well as we are checking only non-org user roles.

File: crates/router/src/core/user.rs [hunk 4]
```diff
@@ -2438,15 +2434,13 @@ pub async fn list_profiles_for_user_in_org_and_merchant_account(
...
                 )
                 .await
                 .change_context(UserErrors::InternalServerError)?
                 .into_iter()
-                .filter_map(|user_role| {
-                    (user_role.status == UserStatus::Active)
-                        .then_some(user_role.profile_id)
-                        .flatten()
-                })
+                .filter_map(|user_role| user_role.profile_id)
```
Comment (ThisIsMani) [inline crates/router/src/core/user.rs:2443]: Here as well.

File: crates/router/src/core/user.rs [hunk 5]
```diff
@@ -2510,12 +2504,13 @@ pub async fn switch_org_for_user(
...
             None,
+            Some(UserStatus::Active),
+            Some(1),
         )
         .await
         .change_context(UserErrors::InternalServerError)
         .attach_printable("Failed to list user roles by user_id and org_id")?
-        .into_iter()
-        .find(|role| role.status == UserStatus::Active)
+        .first()
```
Comment (ThisIsMani) [inline crates/router/src/core/user.rs:2513]: No `to_owned()` needed again.
```suggestion
        .pop()
```

File: crates/router/src/core/user.rs [hunk 6]
```diff
@@ -2712,14 +2707,15 @@ pub async fn switch_merchant_for_user_in_org(
...
+                    Some(1),
                 )
                 .await
                 .change_context(UserErrors::InternalServerError)
                 .attach_printable(
                     "Failed to list user roles for the given user_id, org_id and merchant_id",
                 )?
-                .into_iter()
-                .find(|role| role.status == UserStatus::Active)
+                .first()
```
Comment (ThisIsMani) [inline crates/router/src/core/user.rs:2718]:
```suggestion
                .pop()
```

File: crates/router/src/core/user.rs [hunk 7]
```diff
@@ -2852,6 +2848,8 @@ pub async fn switch_profile_for_user_in_org_and_merchant(
                     Some(&request.profile_id),
                     None,
                     None,
+                    Some(UserStatus::Active),
+                    Some(1)
```
Comment (ThisIsMani) [inline crates/router/src/core/user.rs:2852]: `status` checks are still present after this.

**Review by ThisIsMani — DISMISSED** [review]

**Review by ThisIsMani — APPROVED** [review]

**Review by NishantJoshi00 — APPROVED** [review]

File: crates/diesel_models/src/query/user_role.rs [hunk 8]
```diff
@@ -234,6 +241,14 @@ impl UserRole {
             query = query.filter(dsl::version.eq(version));
         }
 
+        if let Some(status) = status {
+            query = query.filter(dsl::status.eq(status));
+        }
+
+        if let Some(limit) = limit {
+            query = query.limit(limit.into());
```
Comment (lsampras) [inline crates/diesel_models/src/query/user_role.rs:249]: Limit should be compulsory ideally or have a default/max value...

**Pull request merged**